### PR TITLE
Update developer guide on formatting and grammar

### PR DESCRIPTION
**Pull request by koo1993**

### CODE REVIEW

File: docs/DeveloperGuide.adoc [hunk 1]
```diff
@@ -236,11 +236,12 @@ This section describes some noteworthy details on how certain features are imple
 
 The undo/redo mechanism is facilitated by an `UndoRedoStack`, which resides inside `LogicManager`. It supports undoing and redoing of commands that modifies the state of the address book (e.g. `add`, `edit`). Such commands will inherit from `UndoableCommand`.
 
-`UndoRedoStack` only deals with `UndoableCommands`. Commands that cannot be undone will inherit from `Command` instead. The following diagram shows the inheritance diagram for commands:
+`UndoRedoStack` only deals with `UndoableCommands`. Commands that cannot be undone will inherit from `Command` instead. The diagram below (Figure 11) shows the commands inheritance:
```
Comment (shanwpf) [inline docs/DeveloperGuide.adoc]: `The following diagram (Figure 11) shows the inheritance diagram for commands` is better

File: docs/DeveloperGuide.adoc [hunk 2]
```diff
@@ -236,11 +236,12 @@ This section describes some noteworthy details on how certain features are imple
...
 
+.Inheritance Diagram for commands
 image::LogicCommandClassDiagram.png[width="800"]
```
Comment (shanwpf) [inline docs/DeveloperGuide.adoc]: Too cramped, not enough whitespace

File: docs/DeveloperGuide.adoc [hunk 3]
```diff
@@ -236,11 +236,12 @@ This section describes some noteworthy details on how certain features are imple
...
+As you can see from the Figure 11, `UndoableCommand` adds an extra layer between the abstract `Command` class and concrete commands that can be undone, such as the `DeleteCommand`. Note that extra tasks need to be done when executing a command in an _undoable_ way, such as saving the state of the address book before execution. `UndoableCommand` contains the high-level algorithm for those extra tasks while the child classes implement the details of how to execute the specific command. Note that this technique of putting the high-level algorithm in the parent class and lower-level steps of the algorithm in child classes is also known as the https://www.tutorialspoint.com/design_pattern/template_pattern.htm[template pattern].
```
Comment (shanwpf) [inline docs/DeveloperGuide.adoc]: Remove `the` before `Figure 11`

File: docs/DeveloperGuide.adoc [hunk 4]
```diff
@@ -295,16 +296,17 @@ image::UndoRedoExecuteUndoStackDiagram.png[width="800"]
 [NOTE]
 If the `undoStack` is empty, then there are no other commands left to be undone, and an `Exception` will be thrown when popping the `undoStack`.
 
-The following sequence diagram shows how the undo operation works:
+The following sequence diagram (Figure 12) shows how the undo operation works:
```
Comment (shanwpf) [inline docs/DeveloperGuide.adoc]: More whitespace needed around this segment too

File: docs/DeveloperGuide.adoc [hunk 5]
```diff
@@ -317,10 +319,11 @@ The following activity diagram summarize what happens inside the `UndoRedoStack`
 image::UndoRedoActivityDiagram.png[width="650"]
 
 ==== Design Considerations
+This section highlights the implementation consideration of the program.
```
Comment (shanwpf) [inline docs/DeveloperGuide.adoc]: `This section highlights the considerations behind our app's design choices` sounds better

File: docs/DeveloperGuide.adoc [hunk 6]
```diff
@@ -334,28 +337,27 @@ image::UndoRedoActivityDiagram.png[width="650"]
...
-* **Alternative 1 (current choice):** Only include commands that modifies the address book (`add`, `clear`, `edit`).
-** Pros: We only revert changes that are hard to change back (the view can easily be re-modified as no data are * lost).
-** Cons: User might think that undo also applies when the list is modified (undoing filtering for example), * only to realize that it does not do that, after executing `undo`.
+* **Alternative 1 (current choice):** Only include commands that modify the address book (`add`, `clear`, `edit`).
+** Pros: We only revert changes that are hard to change back (the view can easily be re-modified as no data are lost).
+** Cons: User might think that undo also applies when the list is modified (undoing filtering for example),  only to realize that it does not do that, after executing `undo`.
 * **Alternative 2:** Include all commands.
-** Pros: Might be more intuitive for the user.
-** Cons: User have no way of skipping such commands if he or she just want to reset the state of the address * book and not the view.
+** Pros: Program might be more intuitive for the user.
```
Comment (shanwpf) [inline docs/DeveloperGuide.adoc]: Let's standardize and refer to PTMan as app instead of program

File: docs/DeveloperGuide.adoc [hunk 7]
```diff
@@ -334,28 +337,27 @@ image::UndoRedoActivityDiagram.png[width="650"]
...
+** Cons: User has no way of skipping such commands if he or she just want to reset the state of the PTMan and not the view.
```
Comment (shanwpf) [inline docs/DeveloperGuide.adoc]: Remove `the` before `PTMan`

File: docs/DeveloperGuide.adoc [hunk 8]
```diff
@@ -334,28 +337,27 @@ image::UndoRedoActivityDiagram.png[width="650"]
...
 **Additional Info:** See our discussion  https://github.com/se-edu/addressbook-level4/issues/390#issuecomment-298936672[here].
 
-
 ===== Aspect: Data structure to support the undo/redo commands
 
 * **Alternative 1 (current choice):** Use separate stack for undo and redo
-** Pros: Easy to understand for new Computer Science student undergraduates to understand, who are likely to be * the new incoming developers of our project.
-** Cons: Logic is duplicated twice. For example, when a new command is executed, we must remember to update * both `HistoryManager` and `UndoRedoStack`.
+** Pros: Easy to understand for new Computer Science student undergraduates to understand, who are likely to be  the new incoming developers of our project.
```
Comment (shanwpf) [inline docs/DeveloperGuide.adoc]: `Easy to understand for new Computer Science student undergraduates to understand` hmmm

File: docs/DeveloperGuide.adoc (hunk 1 above)
Comment (koo1993) [reply docs/DeveloperGuide.adoc]: Noted with thanks 👍

File: docs/DeveloperGuide.adoc [hunk 9]
```diff
@@ -356,14 +372,14 @@ This section highlights the implementation consideration of the program.
...
-** Cons: User has no way of skipping such commands if he or she just want to reset the state of the PTMan and not the view.
+** Pros: PTMan might be more intuitive for the user.
+** Cons: User has no way of skipping such commands if he or she just want to reset the state of PTMan and not the view.
 **Additional Info:** See our discussion  https://github.com/se-edu/addressbook-level4/issues/390#issuecomment-298936672[here].
 
 ===== Aspect: Data structure to support the undo/redo commands
 
 * **Alternative 1 (current choice):** Use separate stack for undo and redo
-** Pros: Easy to understand for new Computer Science student undergraduates to understand, who are likely to be  the new incoming developers of our project.
+** Pros: Easy to understand for new Computer Science student undergraduates to understand, who are likely to be the new incoming developers of our project.
```
Comment (shanwpf) [inline docs/DeveloperGuide.adoc]: `Easy to understand for new Computer Science student undergraduates to understand` pls

**Pull request merged**